tests/unit/dojson/test_dojson_common_base.py

```diff
@@ -127,153 +123,106 @@ def test_field_from_marcxml_650_with_two_a():
+        '    <subfield code="9">automatically added based on DCC, PPF, DK</subfield>'
         '  </datafield>'
         '</record>'
     )
 
-    expected = [
-        {
-            '_scheme': 'arXiv',
-            'scheme': 'INSPIRE',
-            '_term': 'HEP-PH',
-            'term': 'Phenomenology-HEP',
-        },
-        {
-            '_scheme': 'INSPIRE',
-            'scheme': 'INSPIRE',
-            '_term': 'astro-ph.IM',
-            'term': 'Instrumentation',
-        },
-    ]
-    result = hepnames.do(create_record(snippet))
+    expected_inspire = [{
+        'term': 'Phenomenology-HEP',
+        'source': 'curator',
+    }]
 
-    assert expected == result['field_categories']
+    result = hep.do(create_record(snippet))
+
+    assert expected_inspire == result['inspire_categories']
 
 
-def test_field_from_marcxml_650_with_no_a():
-    """No term at all."""
+def test_field_from_marcxml_650_with_multiple_category():
+    """Two 'a' subfields in one datafield.
+
+    The first is an arXiv fieldcode an the second an INSPIRE category.
+    """
     snippet = (
         '<record>'
         '  <datafield tag="650" ind1="1" ind2="7">'
-        '    <subfield code="2">arXiv</subfield>'
+        '    <subfield code="2">INSPIRE</subfield>'
+        '    <subfield code="a">HEP-PH</subfield>'
+        '    <subfield code="a">astro-ph.CO</subfield>'
+        '    <subfield code="9">arxiv</subfield>'
+        '  </datafield>'
+        '  <datafield tag="650" ind1="1" ind2="7">'
+        '    <subfield code="2">INSPIRE</subfield>'
+        '    <subfield code="a">cs.DL</subfield>'
+        '    <subfield code="9">submitter</subfield>'
         '  </datafield>'
         '</record>'
     )
+    expected_inspire = [{
+        'term': 'Phenomenology-HEP',
+        'source': 'arxiv',
+    },{
+        'term': 'Astrophysics',
+        'source': 'arxiv',
+    },{
+        'term': 'Computing',
+        'source': 'user',
+    }]
 
-    result = hepnames.do(create_record(snippet))
-
-    assert 'field_categories' not in result
-
-
-def test_field_categories_from_650__a_2():
-    snippet = (
-        '<datafield tag="650" ind1="1" ind2="7">'
-        '  <subfield code="2">Inspire</subfield>'
-        '  <subfield code="a">Experiment-HEP</subfield>'
-        '</datafield>'
-    )  # record/1426196
-
-    expected = [
-        {
-            '_scheme': 'Inspire',
-            'scheme': 'INSPIRE',
-            '_term': 'Experiment-HEP',
-            'term': 'Experiment-HEP',
-        },
-    ]
-    result = hepnames.do(create_record(snippet))
-
-    assert expected == result['field_categories']
-
-
-def test_field_categories_from_65017_a_2_9_with_conference():
-    snippet = (
-        '<datafield tag="650" ind1="1" ind2="7">'
-        '  <subfield code="2">INSPIRE</subfield>'
-        '  <subfield code="9">conference</subfield>'
-        '  <subfield code="a">Accelerators</subfield>'
-        '</datafield>'
-    )  # record/1479228
```

Comment:
Be careful: you should *never* delete tests that use real data in favor of tests that use fake data. These tests are here because they were documenting a regression; if you delete them you allow the regression to happen again.

Sometimes you have to, because the business logic changed so much that the past test is not salvageable (but it's not the case here). In this case, _explain_ why it's safe to delete the test!